graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt

```diff
@@ -4832,4 +4832,36 @@ It takes a title and such.
         }
         assertThat(exception.message).isEqualTo("Property \"firstname\" does not exist in input type \"Person\"")
     }
+
+    @Test
+    fun `The default value for BigDecimal should be overridden`() {
```

Comment:
Thank you for adding tests! Would you mind adding a case that would exercise the `null` (or exception if you make that change) route?

Reply:
Yes, I'll add tests to cover handling of incompatible values 😉